lib/matplotlib/axes/_base.py

```diff
@@ -3492,10 +3492,22 @@ def set_xlabel(self, xlabel, fontdict=None, labelpad=None, *,
             loc = (loc if loc is not None
                    else mpl.rcParams['xaxis.labellocation'])
         _api.check_in_list(('left', 'center', 'right'), loc=loc)
+        kwargs = cbook.normalize_kwargs(kwargs, mtext.Text)
+        x = kwargs.get('x')
+        ha = kwargs.get('horizontalalignment')
+
         if loc == 'left':
-            kwargs.update(x=0, horizontalalignment='left')
+            kwargs.update(x=0 if x is None else x,
+                          horizontalalignment='left' if ha is None else ha)
```

Comment:
I think this would look better as two `setdefault` calls:
```suggestion
            kwargs.setdefault('x', 0)
            kwargs.setdefault('horizontalalignment', 'left')
```
and the same in the remaining.